components/scream/src/physics/cld_fraction/atmosphere_cld_fraction.cpp

```diff
@@ -0,0 +1,135 @@
+#include "atmosphere_cld_fraction.hpp"
+
+#include "ekat/ekat_assert.hpp"
+#include "ekat/util/ekat_units.hpp"
+
+#include <array>
+
+namespace scream
+{
+  using namespace cld_fraction;
+// =========================================================================================
+CldFraction::CldFraction (const ekat::Comm& comm, const ekat::ParameterList& params)
+ : m_cldfraction_comm (comm)
+ , m_cld_fraction_params (params)
+{
+  // Nothing to do here
+}
+
+// =========================================================================================
+void CldFraction::set_grids(const std::shared_ptr<const GridsManager> grids_manager)
+{
+  using namespace ekat::units;
+
+  // The units of mixing ratio Q are technically non-dimensional.
+  // Nevertheless, for output reasons, we like to see 'kg/kg'.
+  auto Q = kg/kg;
+  Q.set_string("kg/kg");
+  Units nondim(0,0,0,0,0,0,0);
+
+  const auto& grid_name = m_cld_fraction_params.get<std::string>("Grid");
+  auto grid  = grids_manager->get_grid(grid_name);
+  m_num_cols = grid->get_num_local_dofs(); // Number of columns on this rank
+  m_num_levs = grid->get_num_vertical_levels();  // Number of levels per column
+
+  // Define the different field layouts that will be used for this process
+  using namespace ShortFieldTagsNames;
+
+  // Layout for 3D (2d horiz X 1d vertical) variable defined at mid-level and interfaces 
+  FieldLayout scalar3d_layout_mid { {COL,LEV}, {m_num_cols,m_num_levs} };
+
+  // Set of fields used strictly as input
+  m_required_fields.emplace("qi",   scalar3d_layout_mid, Q,      grid_name);
+  m_required_fields.emplace("alst", scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used strictly as output
+  m_computed_fields.emplace("ast",   scalar3d_layout_mid, nondim, grid_name);
+  m_computed_fields.emplace("aist",  scalar3d_layout_mid, nondim, grid_name);
+
+  // Set of fields used as input and output
+  // - There are no fields used as both input and output.
+
+}
+
+// =========================================================================================
+void CldFraction::initialize_impl (const util::TimeStamp& t0)
```

Comment:
Since `t0` is not used, you should comment it, to avoid compiler warnings. Simply put it inside comments, like `/* t0 */`.